index.html

```diff
@@ -389,9 +393,48 @@ <h2>
                     <var>message</var>)</code>.</p>
                   </li>
                   <li>
-                    <p><a>Resolve</a> <var>p</var> with <var>stream</var> and
-                    abort these steps. This invocation of {{MediaDevices/getDisplayMedia()}}
-                    is now considered to have produced a new <dfn>capture-session</dfn></p>
+                    <p>
+                      This invocation of {{MediaDevices/getDisplayMedia()}} is now
+                      considered to have produced a new <dfn>capture-session</dfn>.
+                    </p>
+                  </li>
+                  <li>
+                    If <var>controller</var> is not <code>null</code>:
+                    <ol>
+                      <li>
+                        <p>
+                          Set <var>controller</var>.{{CaptureController/[[Source]]}}
+                          to <var>stream</var>'s video track's <a data-cite="GETUSERMEDIA#dfn-source-0">[[\Source]]</a>.
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          Set <var>controller</var>.{{CaptureController/[[DisplaySurfaceType]]}} to the 
+                          to <var>stream</var>'s video track's {{DisplayCaptureSurfaceType}}.
+                        </p>
+                      </li>
+                      <li>
+                        <p>
+                          Schedule a <dfn>task to close the window of opportunity to set focus behavior</dfn>,
+                          which will set <var>controller</var>.{{CaptureController/[[FocusWindowClosed]]}}
+                          to <code>true</code> when it executes.
```

Comment:
We need to use "[queue a task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-task)" language here. This spec has languished a bit, and we should update it at some point to match modern specs more (e.g. what task source are we using?), but small steps.

I wouldn't make this step into a definition since it doesn't seem to support a normative reference from another algorithm, nor does it seem like a well-defined concept.

I like "window of opportunity", but worry the metaphors may be a bit on the nose and confusing in this context. How about `[[FocusOpportunityExpired]]`?

```suggestion
                          Queue a task to set <var>controller</var>.{{CaptureController/[[FocusOpportunityExpired]]}}
                          to <code>true</code>.
```

Reply:
I'm fine with `[[FocusOpportunityExpired]]`.